src/AzureClient/AzureClient.cs

```diff
@@ -77,98 +69,33 @@ private string ValidExecutionTargetsDisplayText
-            var cacheHelper = await MsalCacheHelper.CreateAsync(storageCreationProperties);
-            cacheHelper.RegisterCache(msalApp.UserTokenCache);
-
-            bool shouldShowLoginPrompt = refreshCredentials;
-            if (!shouldShowLoginPrompt)
-            {
-                try
-                {
-                    var accounts = await msalApp.GetAccountsAsync();
-                    AuthenticationResult = await msalApp.AcquireTokenSilent(
-                        azureEnvironment.Scopes, accounts.FirstOrDefault()).WithAuthority(msalApp.Authority).ExecuteAsync();
-                }
-                catch (MsalUiRequiredException)
-                {
-                    shouldShowLoginPrompt = true;
-                }
-            }
-
-            if (shouldShowLoginPrompt)
-            {
-                AuthenticationResult = await msalApp.AcquireTokenWithDeviceCode(
-                    azureEnvironment.Scopes,
-                    deviceCodeResult =>
-                    {
-                        channel.Stdout(deviceCodeResult.Message);
-                        return Task.FromResult(0);
-                    }).WithAuthority(msalApp.Authority).ExecuteAsync();
-            }
-
-            if (AuthenticationResult == null)
+            var azureEnvironment = AzureEnvironment.Create(subscriptionId);
+            ActiveWorkspace = await azureEnvironment.GetAuthenticatedWorkspaceAsync(channel, resourceGroupName, workspaceName, refreshCredentials);
+            if (ActiveWorkspace == null)
             {
                 return AzureClientError.AuthenticationFailed.ToExecutionResult();
             }
 
-            var credentials = new Rest.TokenCredentials(AuthenticationResult.AccessToken);
-            QuantumClient = new QuantumClient(credentials)
-            {
-                SubscriptionId = subscriptionId,
-                ResourceGroupName = resourceGroupName,
-                WorkspaceName = workspaceName,
-                BaseUri = azureEnvironment.BaseUri,
-            };
-            ActiveWorkspace = new Azure.Quantum.Workspace(
-                QuantumClient.SubscriptionId,
-                QuantumClient.ResourceGroupName,
-                QuantumClient.WorkspaceName,
-                AuthenticationResult?.AccessToken,
-                azureEnvironment.BaseUri);
-
-            try
-            {
-                AvailableProviders = await QuantumClient.Providers.GetStatusAsync();
-            }
-            catch (Exception e)
+            AvailableProviders = await ActiveWorkspace.GetProvidersAsync();
+            if (AvailableProviders == null)
             {
-                Logger?.LogError(e, $"Failed to download providers list from Azure Quantum workspace: {e.Message}");
                 return AzureClientError.WorkspaceNotFound.ToExecutionResult();
             }
```

Comment:
This should be a different error, as there is a workspace...

Reply:
Yes, we have created the "workspace" object on the client, but downloading the providers list is the first time we actually attempt to communicate with Azure Quantum. For example, if the user has mistyped the workspace name, this is the error they will see. The friendly string displayed is: "No Azure Quantum workspace was found that matches the specified criteria."